api/mixins.py

```diff
@@ -0,0 +1,42 @@
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from api.serializers import UserSerializer
+from likes import services
+from likes.models import LikeDislike
+
+
+class LikedMixin:
+    @action(methods=['POST'], detail=True, url_path='vote/(?P<vote_type>\\w+)')
+    def vote(self, request, pk=None, vote_type=None):
+        """Likes or dislikes obj depending on the type of voice (like or dislike).
+        """
+        votes = {'like': LikeDislike.LIKE, 'dislike': LikeDislike.DISLIKE}
+        obj = self.get_object()
+        try:
+            services.add_likedislike(obj, request.user, votes[vote_type])
+        except KeyError:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        return Response()
+
+    @action(methods=['POST'], detail=True)
```

Comment:
issue: так доступно только админу
```suggestion
    @action(methods=['POST'], detail=True, permission_classes=(IsAuthenticated,),)
```